src/tests/ftest/container/rebuild_tests.py

```diff
@@ -0,0 +1,175 @@
+
+    def test_rebuild_cont_create(self):
+        """
+        Test Description: Test creating a container while rebuild is ongoing.
+
+        :avocado: tags=cont,rebuild,rebuildsimple
+        """
+        try:
+
+            # make ourselves a pool
+            pool = DaosPool(self.context)
+            pool.create(self.createmode, self.createuid, self.creategid,
+                        self.createsize, self.createsetid)
+
+            # want an open connection during rebuild
+            pool.connect(1 << 1)
+
+            # get pool status we want to test later
+            pool.pool_query()
+            if pool.pool_info.pi_ndisabled != 0:
+                self.fail("Number of disabled targets reporting incorrectly.")
+            if pool.pool_info.pi_rebuild_st.rs_errno != 0:
+                self.fail("Rebuild error but rebuild hasn't run.")
+            if pool.pool_info.pi_rebuild_st.rs_obj_nr != 0:
+                self.fail("Rebuilt objs not zero.")
+            if pool.pool_info.pi_rebuild_st.rs_rec_nr != 0:
+                self.fail("Rebuilt recs not zero.")
+
+            # create and open a container
+            container = DaosContainer(self.context)
+            container.create(pool.handle)
+            container.open()
+
+            # fill the pool with enough data such that rebuild runs for a bit
+            saved_data = []
+            for _objc in range(self.objcount):
+                obj = None
+                for _recc in range(self.reccount):
+                    dkey = (
+                        ''.join(random.choice(string.ascii_uppercase +
+                                              string.digits) for _ in range(5)))
+                    akey = (
+                        ''.join(random.choice(string.ascii_uppercase +
+                                              string.digits) for _ in range(5)))
+                    data = (''.join(random.choice(string.ascii_uppercase +
+                                                  string.digits) for _ in
+                                    range(self.size)))
+                    obj, txn = container.write_an_obj(data, len(data), dkey,
+                                                      akey, obj, self.rank,
+                                                      obj_cls=16)
+                    saved_data.append((obj, dkey, akey, data, txn))
+
+                    # validate the written object
+                    data2 = container.read_an_obj(self.size, dkey, akey, obj,
+                                                  txn)
+                    if data != data2.value:
+                        self.fail("Write data 1, read it back, didn't match\n")
+
+            # kill a server
+            server = DaosServer(self.context, self.server_group, self.rank)
+            server.kill(1)
+
+            # temporarily, the exclude of a failed target must be done manually
+            pool.exclude([self.rank])
+
+            while True:
```

Comment:
I'm not clear on why a loop is needed here?  It will never loop, either exit early with the fail statement or will hit the break.

Reply:
See Phil's comment above, it's attempting to solve that problem. /pool/rebuild_tests implements a similar mechanism. I have removed the loop but with the same problem; as Phil pointed out, there's a potential race there.